### PR TITLE
Optimization on the process of receiving in quic stream

### PR DESCRIPTION
**Pull request by wanghaEMQ**
* Optimization on the process of receiving in quic stream
* Fix some warnings in quic_api.c

### CODE REVIEW

File: src/supplemental/quic/quic_api.c [hunk 1]
```diff
@@ -1455,6 +1457,7 @@ quic_pipe_open(void *qsock, void **qpipe, void *mqtt_pipe)
 int
 quic_pipe_close(void *qpipe, uint8_t *code)
 {
+	(void) code;
```
Comment (RanMaoyi) [inline src/supplemental/quic/quic_api.c:1460]: Maybe u need this
`NNI_ARG_UNUSED(code)`

**Review by RanMaoyi — APPROVED** [review]
LGTM

File: src/supplemental/quic/quic_api.c (hunk 1 above)
Comment (wanghaEMQ) [reply src/supplemental/quic/quic_api.c:1460]: Yes. u are right

**Review by lee-emqx — APPROVED** [review]

**Review by JaylinYu — APPROVED** [review]

**Comment by JaylinYu** [comment]
Needs a deep benchmark testing

**Pull request merged**